azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -201,15 +215,44 @@ private boolean isMultiExecutorMode() {
    */
   private void refreshExecutors() {
     synchronized (activeExecutors) {
-      // TODO: rest api call to refresh executor stats
+      ExecutorService taskExecutors =
```

Comment:
Thread pool is a bit expensive, is it possible to reuse the taskExecutors thread pool?  any reason why you don't want to reuse?

This method can just submit refresh tasks.